lib/iris/cube.py

```diff
@@ -3575,23 +3575,23 @@ def __eq__(self, other):
                 )
 
             if result:
-                coord_comparison = iris.analysis.coord_comparison(
+                comparison = iris.analysis._dimensional_metadata_comparison(
```

Comment:
@stephenworsley Seems like we're baking in some indirection here, where the behaviour of our public API is now depending on the private behaviour of `_dimensional_metadata_comparison`. 

This means that indirectly we can change the public API behaviour of `iris.cube.Cube.__eq__` by changing `iris.analysis.__init__._dimensional_metadata_comparison`... that seems a tad dangerous to me.

Reply:
> we can change the public API behaviour of iris.cube.Cube.__eq__ by changing iris.analysis.__init__._dimensional_metadata_comparison

I don't see why this is a particular concern.  It should be obvious that we *wouldn't* change the behaviour of the public function without very careful considering + management, and I think the public API dependence on the private function should be clear enough to any potential developer that it wouldn't happen by accident.

Of course, if we make the more general private function public all such problems would go away.  But IMHO that seems like a good deal more work that we just don't really need to commit to at this stage.